libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -40,12 +50,16 @@ public BotFrameworkAdapter Use(Middleware.IMiddleware middleware)
             return this;
         }
 
-        public async Task ProcessActivty(string authHeader, Activity activity, Func<IBotContext, Task> callback)
+        public async Task ProcessActivty(string authHeader, IActivity activity, Func<IBotContext, Task> callback)
         {
             BotAssert.ActivityNotNull(activity);
-            await JwtTokenValidation.AssertValidActivity(activity, authHeader, _credentialProvider, _httpClient);
+            ClaimsIdentity claimsIdentity =  await JwtTokenValidation.EnsureValidActivity(activity, authHeader, _credentialProvider, _httpClient);
 
-            var context = new BotContext(this, activity);
+            // For requests from channel App Id is in Audience claim of JWT token. For emulator it is in AppId claim. For 
+            // unauthenticated requests we have anonymouse identity provided auth is disabled.
+            string botAppId = (claimsIdentity.Claims?.SingleOrDefault(claim => claim.Type == AuthenticationConstants.AudienceClaim) ??
```

Comment:
This code is just extracting AppId and adding it to context. This could be used by a bot for various other scenarios e.g. accessing Graph (using the same AADv2 App as the bot). Also extracting this anywhere else means Auth header needs to be reprocessed.